frontend/public/components/network-policy.tsx

```diff
@@ -135,60 +135,82 @@ const EgressHeader = () => {
-  let i = 0;
-
   const style = { margin: '5px 0' };
-  _.each(peers, ({ namespaceSelector, podSelector: ps }) => {
-    if (namespaceSelector) {
-      nsSelectors.push(
-        <div key={i++} style={style}>
-          <Selector selector={namespaceSelector} kind="Namespace" />
-        </div>,
-      );
-    } else {
-      podSelectors.push(
-        <div key={i++} style={style}>
-          <Selector selector={ps} namespace={namespace} />
-        </div>,
-      );
-    }
-  });
+
   return (
     <div className="row co-resource-list__item">
       <div className="col-xs-4">
         <div>
           <span className="text-muted">{t('public~Pod selector')}:</span>
         </div>
         <div style={style}>
-          <Selector selector={podSelector} namespace={namespace} />
+          <Selector selector={mainPodSelector} namespace={namespace} />
         </div>
       </div>
       <div className="col-xs-5">
         <div>
-          {!podSelectors.length ? null : (
-            <div>
-              <span className="text-muted">{t('public~Pod selector')}:</span>
-              {podSelectors}
-            </div>
-          )}
-          {!nsSelectors.length ? null : (
-            <div style={{ paddingTop: podSelectors.length ? 10 : 0 }}>
-              <span className="text-muted">{t('public~NS selector')}:</span>
-              {nsSelectors}
-            </div>
+          {peer ? (
+            <>
+              {peer.namespaceSelector ? (
+                <div>
+                  <span className="text-muted">{t('public~NS selector')}:</span>
+                  <div style={style}>
+                    {_.isEmpty(peer.namespaceSelector) ? (
+                      <span>{t('public~Any namespace')}</span>
+                    ) : (
+                      <Selector selector={peer.namespaceSelector} kind="Namespace" />
+                    )}
+                  </div>
+                </div>
+              ) : (
+                peer.podSelector && (
+                  <div>
+                    <span className="text-muted">{t('public~Namespace')}:</span>
+                    <div style={style}>{namespace}</div>
+                  </div>
+                )
+              )}
+              {peer.podSelector && (
+                <div style={{ paddingTop: 10 }}>
+                  <span className="text-muted">{t('public~Pod selector')}:</span>
+                  <div style={style}>
+                    {_.isEmpty(peer.podSelector) ? (
+                      <span>{t('public~Any pod')}</span>
+                    ) : (
+                      <Selector
+                        selector={peer.podSelector}
+                        namespace={peer.namespaceSelector ? undefined : namespace}
+                      />
+                    )}
+                  </div>
+                </div>
+              )}
+              {peer.ipBlock && (
+                <div>
+                  <span className="text-muted">{t('public~IP Block')}:</span>
```

Comment:
We should use sentence case for labels

```suggestion
                  <span className="text-muted">{t('public~IP block')}:</span>
```